code/modules/clothing/neck/neck.dm

```diff
@@ -148,11 +148,91 @@
 	icon_state = "petcollar"
 	item_color = "petcollar"
 	var/tagname = null
+	var/obj/item/card/id/id
 
 /obj/item/clothing/neck/petcollar/attack_self(mob/user)
 	tagname = copytext(sanitize(input(user, "Would you like to change the name on the tag?", "Name your new pet", "Spot") as null|text),1,MAX_NAME_LEN)
 	name = "[initial(name)] - [tagname]"
 
+/obj/item/clothing/neck/petcollar/proc/remove_id()
+	if (id)
+		if (ismob(loc))
+			var/mob/M = loc
+			M.put_in_hands(id)
+			to_chat(usr, "<span class='notice'>You remove the ID from the [name].</span>")
+		else
+			id.forceMove(get_turf(src))
+		id = null
+
+/obj/item/clothing/neck/petcollar/verb/verb_remove_id()
+	set category = "Object"
+	set name = "Remove ID"
+	set src in usr
+
+	if(issilicon(usr))
+		return
+
+	if (usr.canUseTopic(src))
+		if(id)
+			remove_id()
+		else
+			to_chat(usr, "<span class='warning'>This collar does not have an ID in it!</span>")
+
+/obj/item/clothing/neck/petcollar/proc/id_check(mob/user, obj/item/card/id/I)
+	if(!I)
+		if(id)
+			remove_id()
+			return TRUE
+		else
+			var/obj/item/card/id/C = user.get_active_held_item()
+			if(istype(C))
+				I = C
+
+	if(I)
+		if(!user.transferItemToLoc(I, src))
+			return FALSE
+		var/obj/old_id = id
+		id = I
+		if(old_id)
+			user.put_in_hands(old_id)
```

Comment:
what if this fails?

Reply:
Think I just fixed.

Reply:
no it still doesn't move old_id anywhere if it can't put in hands

Reply:
Fixing.